client_integration_test.go

```diff
@@ -1249,55 +1249,50 @@ func TestClientReadSequential(t *testing.T) {
 	}
 }
 
+// this writer requires maxPacket = 3 and always returns an error for the second write call
 type lastChunkErrSequentialWriter struct {
-	expected      int
-	written       int
-	writtenReturn int
+	writeCounter int
```

Comment:
[nit] we don’t really need the `write` of the `writeCounter`. This is a writer after all.